### PR TITLE
De-hardcode database implementation

### PR DESCRIPTION
**Pull request by l-jonas**
This is a breaking change because the users of the library have to manually depend on the classic repository implementation and supply it manually.

### CODE REVIEW

File: bep/src/main/kotlin/net/syncthing/java/bep/IndexBrowser.kt [hunk 1]
```diff
@@ -129,7 +129,7 @@ class IndexBrowser internal constructor(private val indexRepository: IndexReposi
 
     fun listFiles(path: String = currentPath): List<FileInfo> {
         logger.debug("doListFiles for path = '{}' BEGIN", path)
-        val list = indexRepository.findNotDeletedFilesByFolderAndParent(folder, path)
+        val list = ArrayList(indexRepository.findNotDeletedFilesByFolderAndParent(folder, path))
```
Comment (Nutomic) [inline bep/src/main/kotlin/net/syncthing/java/bep/IndexBrowser.kt:132]: I think it's better to use `.toList()` here. This version might actually give you an `ArrayList(Set())` or something like that.

File: client/src/main/kotlin/net/syncthing/java/client/Main.kt [hunk 2]
```diff
@@ -13,6 +13,13 @@
  */
 package net.syncthing.java.client
 
+// this is archived
+//
+// I'm not sure if anyone still needs it. If someone needs it,
+// it should be moved to an other package because the SqlRepository
+// should not be a dependency of this module.
+
+/*
```
Comment (Nutomic) [inline client/src/main/kotlin/net/syncthing/java/client/Main.kt]: I think this is mainly intended for testing. Using a command line tool for that is much easier than compiling an Android app.

File: bep/src/main/kotlin/net/syncthing/java/bep/IndexBrowser.kt (hunk 1 above)
Comment (l-jonas) [reply bep/src/main/kotlin/net/syncthing/java/bep/IndexBrowser.kt:132]: It's already an [list](https://github.com/syncthing/syncthing-java/pull/18/files#diff-b26f9e3d5d897a1dc34f709b125658a3R40), but as it's not mutable and [sometimes an new entry is added](https://github.com/syncthing/syncthing-java/blob/c263ed8c45cf819fbc577c2c9cbb8cfed0c51080/bep/src/main/kotlin/net/syncthing/java/bep/IndexBrowser.kt#L135), it's required to make it a mutable list.

The original implementation [returns a MutableList](https://github.com/syncthing/syncthing-java/blob/master/repository/src/main/kotlin/net/syncthing/java/repository/repo/SqlRepository.kt#L398), but the new one does not and it's a good practice to clone data from somewhere else before modifying it (assume that there could be a caching in the implementation).

Comment (Nutomic) [reply bep/src/main/kotlin/net/syncthing/java/bep/IndexBrowser.kt:132]: Ah good. Turns out I already forgot the little bit I learned about Kotlin :D

**Pull request merged**